unified_planning/plans/sequential_plan.py

```diff
@@ -0,0 +1,104 @@
+        return str(self._actions)
+
+    def __eq__(self, oth: object) -> bool:
+        if isinstance(oth, SequentialPlan) and len(self._actions) == len(oth._actions):
+            for ai, oth_ai in zip(self._actions, oth._actions):
+                if ai.action != oth_ai.action or ai.actual_parameters != oth_ai.actual_parameters:
+                    return False
+            return True
+        else:
+            return False
+
+    @property
+    def actions(self) -> List['plans.plan.ActionInstance']:
+        '''Returns the sequence of action instances.'''
+        return self._actions
+
+    def replace_action_instances(self, replace_function: Callable[['plans.plan.ActionInstance'], 'plans.plan.ActionInstance']) -> 'up.plans.plan.Plan':
+        return SequentialPlan([replace_function(ai) for ai in self._actions])
+
+    def to_partial_order_plan(self, _env: Optional['up.Environment'] = None) -> 'up.plans.partial_order_plan.PartialOrderPlan':
+        env = up.environment.get_env(_env)
+        subs = Substituter(env)
+        simp = Simplifier(env)
+        fve = FreeVarsExtractor()
+        # last_modifier is the mapping from a grounded fluent to the last action instance that assigned a value to
+        # that fluent
+        last_modifier: Dict[FNode, 'plans.plan.ActionInstance'] = {}
+        # all_required is the mapping from a grounded fluent to all the action instances that read the value of that
+        # fluent in their preconditions (or in the condition of their conditional effects)
+        all_required: Dict[FNode, List['plans.plan.ActionInstance']] = {}
+        # graph stores the information gathered through the process
+        graph = nx.DiGraph()
+        for action_instance in self.actions:
+            grounded_action = cast(InstantaneousAction, up.plans.plan.ground_action_instance(action_instance, subs, simp))
+            graph.add_node(action_instance)
+
+            # required_fluents contains all the fluents that this action_instance "reads"
+            required_fluents: Set[FNode] = set()
+            # add free vars of preconditions
+            for prec in grounded_action.preconditions:
+                required_fluents = required_fluents.union(fve.get(prec))
+            # add in the required_fluents all the free fluents this action instance deals with
+            for eff in grounded_action.effects:
+                required_fluents = required_fluents.union(fve.get(eff.condition))
+                required_fluents = required_fluents.union(fve.get(eff.fluent))
+                required_fluents = required_fluents.union(fve.get(eff.value))
+
+            # for every required fluent, add this action instance to the list of action instances that requires this fluent
+            # and order the current action instance after the last modifier of the fluent
+            for required_fluent in required_fluents:
+                action_instance_list = all_required.get(required_fluent, None)
```

Comment:
You can be smart and do 
```suggestion
                action_instance_list = all_required.setdefault(required_fluent, [])
```
so that you do not need the switch below